Source/DafnyCore/Compilers/Compiler-python.cs

```diff
@@ -853,14 +865,21 @@ private class ClassWriter : IClassWriter {
     }
 
     protected override void EmitHalt(IToken tok, Expression messageExpr, ConcreteSyntaxTree wr) {
+      Contract.Requires(tok != null);
       var wStmts = wr.Fork();
       wr.Write($"raise {DafnyRuntimeModule}.HaltException(");
-      if (tok != null) {
-        wr.Write($"\"{ErrorReporter.TokenToString(tok)}: \" + {DafnyRuntimeModule}.string_of(");
+      wr.Write($"\"{ErrorReporter.TokenToString(tok)}: \" + ");
+
+      if (UnicodeChars && messageExpr.Type.IsStringType) {
+        TrParenExpr(messageExpr, wr, false, wStmts);
+        wr.Write(".VerbatimString(False)");
+      } else {
+        wr.Write($"{DafnyRuntimeModule}.string_of(");
+        TrExpr(messageExpr, wr, false, wStmts);
+        wr.WriteLine(")");
       }
```

Comment:
Let's share this code with `EmitPrintStmt` above

Reply:
Good call, it's interesting that only the Java compiler has `EmitToString` when both `print` and `expect` implicitly depend on that operation. I might even define that for all compilers.

Reply:
No that's too aggressive, there's too much of the "best effort to track strings at runtime" logic in the other compilers.